cli/cmd/setup.go

```diff
@@ -106,14 +44,67 @@ func appendValueToPrompt(promptString, value string) string {
 
 func init() {
 	rootCmd.AddCommand(setupCmd)
+}
+
+func runSetup(cmd *cobra.Command, args []string) {
+	var zabbixURL, zabbixUser, zabbixPassword, rundeckURL, rundeckAPIKey string
+	zabbixConfigKeyName := "zabbix"
+	rundeckConfigKeyName := "rundeck"
+
+	// Read Zabbix config
+
+	if viper.IsSet(zabbixConfigKeyName) {
+		zabbixURL = getConfigValueByKey(zabbixConfigKeyName, "url")
+		zabbixUser = getConfigValueByKey(zabbixConfigKeyName, "username")
+		zabbixPassword = getConfigValueByKey(zabbixConfigKeyName, "password")
+	}
+
+	fmt.Println(appendValueToPrompt("Enter the Zabbix Server URL:", zabbixURL))
+	fmt.Scanln(&zabbixURL)
+
+	fmt.Println(appendValueToPrompt("Enter the Zabbix Username:", zabbixUser))
+	fmt.Scanln(&zabbixUser)
 
-	// Here you will define your flags and configuration settings.
+	fmt.Println(appendValueToPrompt("Enter the Zabbix Password:", zabbixPassword))
+	fmt.Scanln(&zabbixPassword)
 
-	// Cobra supports Persistent Flags which will work for this command
-	// and all subcommands, e.g.:
-	// setupCmd.PersistentFlags().String("foo", "", "A help for foo")
+	// Generate Zabbix API key
+	z, err := zabbix.NewAPI(zabbixURL, "", zabbixUser, zabbixPassword)
+	zabbixAPIKey, err := z.GetKey()
+
+	if err != nil {
+		log.Errorf("cannot generate Zabbix API key. Error: %v", err)
+		return
+	}
+
+	// Read Rundeck config
+	if viper.IsSet(rundeckConfigKeyName) {
+		rundeckURL = getConfigValueByKey(rundeckConfigKeyName, "url")
+		rundeckAPIKey = getConfigValueByKey(rundeckConfigKeyName, "api_key")
+	}
+	fmt.Println(appendValueToPrompt("Enter the Rundeck URL:", rundeckURL))
+	fmt.Scanln(&rundeckURL)
+
+	fmt.Println(appendValueToPrompt("Enter the Rundeck API_KEY:", rundeckAPIKey))
+	fmt.Scanln(&rundeckAPIKey)
+
+	zabbixConfig := lib.ZabbixConfig{
+		URL:      zabbixURL,
+		UserName: zabbixUser,
+		Password: zabbixPassword,
+		APIKey:   zabbixAPIKey,
+	}
+
+	rundeckConfig := lib.RundeckConfig{
+		URL:    rundeckURL,
+		APIKey: rundeckAPIKey,
+	}
+
+	newConfig := lib.Config{
+		Zabbix:  zabbixConfig,
+		Rundeck: rundeckConfig,
+	}
 
-	// Cobra supports local flags which will only run when this command
-	// is called directly, e.g.:
-	// setupCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	newConfig.Save()
```

Comment:
This seems to be ignoring error saving the config.
You can just return the error to the caller.